test/unit/test_qpu.py

```diff
@@ -21,3 +54,48 @@ def test_provided_execution_options():
 
     qpu = QPU(quantum_processor_id="test", execution_options=options)
     assert qpu.execution_options == options
+
+
+@patch("pyquil.api._qpu.retrieve_results")
+@patch("pyquil.api._qpu.submit")
+def test_qpu_execute(mock_submit, mock_retrieve_results, mock_encrypted_program):
```

Comment:
I might be inconsistent in calling for this, but please type test params for the benefit of later maintenance